tektoncd/pipelines-as-code
# Update golangci-lint, sync go-toolset/vale/pre-commit

# Changes

- Update go-toolset image to latest
- Update vale styles to latest
- Enable new golangci-lint rules
  Enable a bunch of new golangci-lint, commented the ones we don't use so
  to make it easier to get new linters in the future.
  Corrected a lot of errors
- Update pre-commit hooks
- Refactor getInstallationIDFromPayload function
  Don't do twice marshal/unmarshal in the function when it's not needed
- Remove black from linter.yaml and use ruff format to format check
  Check if the files are formatted with ruff format --check instead of
  black since we know use it for linting we may as well use it for
  formatting (and faster).





# Submitter Checklist

- [X] 📝 A good commit message is important for other reviewers to understand the context of your change. Please refer to [How to Write a Git Commit Message](https://cbea.ms/git-commit/) for more details how to write beautiful commit messages. We rather have the commit message in the PR body and the commit message instead of an external website.
- [X] ♽ Run `make test` before submitting a PR (ie: with [pre-commit](https://pipelinesascode.com/dev/tools), no need to waste CPU cycle on CI. (or even better install [pre-commit](https://pre-commit.com/) and do `pre-commit install` in the root of this repo).
- [X] ✨ We heavily rely on linters to get our code clean and consistent, please ensure that you have run `make lint` before submitting a PR. The [markdownlint](https://github.com/DavidAnson/markdownlint) error can get usually fixed by running `make fix-markdownlint` (make sure it's installed first)
- [X] 📖 If you are adding a user facing feature or make a change of the behavior, please verify that you have documented it
- [X ] 🧪 100% coverage is not a target but most of the time we would rather have a unit test if you make a code change.
- [X] 🎁 If that's something that is possible to do please ensure to check if we can add a e2e test.
- [X] 🔎 If there is a flakiness in the CI tests then don't *necessary* ignore it, better get the flakyness fixed before merging or if that's not possible there is a good reason to bypass it. (token rate limitation may be a good reason to skip).
